docs/side_quests/ide_features.md

```diff
@@ -0,0 +1,654 @@
+
+## What you'll learn here
+
+This guide focuses on **IDE productivity features** that will make you a more efficient Nextflow developer:
+
+- **Advanced syntax highlighting**: Understanding what VS Code is showing you about your code structure
+- **Intelligent auto-completion**: Leveraging context-aware suggestions for faster code writing
+- **Error detection and diagnostics**: Catching syntax errors before you run your workflow
+- **Code navigation**: Quickly moving between processes, modules, and definitions
+- **Formatting and organization**: Maintaining consistent, readable code style
+- **AI-assisted development** (optional): Using modern AI tools integrated with your IDE
+
+!!! info "Why IDE features now?"
+
+    You've likely already been using VS Code during the [Hello Nextflow](../hello_nextflow/) course, but we kept the focus on learning Nextflow fundamentals rather than IDE features. Now that you're comfortable with basic Nextflow concepts like processes, workflows, channels, and modules, you're ready to leverage the sophisticated IDE features that will make you a more efficient developer.
+
+    Think of this as "leveling up" your development environment - the same editor you've been using has much more powerful capabilities that become truly valuable once you understand what they're helping you with.
+
+---
+
+## 0. Setup and Warmup
+
+Let's set up a workspace specifically for exploring IDE features:
+
+```bash title="Navigate to the IDE features directory"
+cd side-quests/ide_features
+```
+
+Open this directory in VS Code:
+
+```bash title="Open VS Code in current directory"
+code .
+```
+
+The `ide_features` directory contains example workflows that demonstrate various IDE features:
+
+```bash title="Show directory structure"
+tree .
+```
+
+```console title="Project structure"
+tree .
+.
+├── basic_workflow.nf
+├── complex_workflow.nf
+├── data
+│   ├── sample_001.fastq.gz
+│   ├── sample_002.fastq.gz
+│   ├── sample_003.fastq.gz
+│   ├── sample_004.fastq.gz
+│   ├── sample_005.fastq.gz
+│   └── sample_data.csv
+├── modules
+│   ├── fastqc.nf
+│   ├── star.nf
+│   └── utils.nf
+└── nextflow.config
+
+3 directories, 12 files
+```
+
+!!! note "About the Example Files"
+
+    - `basic_workflow.nf` is a working basic workflow that you can run and modify
+    - `complex_workflow.nf` is designed for illustration only to demonstrate navigation features - it may not run successfully but shows realistic multi-file workflow structure
+
+### 0.1. Installing the Nextflow Extension
+
+!!! note "Already Using Devcontainers?"
+
+    If you're working in **GitHub Codespaces** or using a **local devcontainer**, the Nextflow extension is likely already installed and configured for you. You can skip the manual installation steps below and proceed directly to exploring the extension features.
+
+To install the extension manually:
+
+1. Open VS Code
+2. Go to the Extensions view (`Ctrl+Shift+X` or `Cmd+Shift+X`)
```

Comment:
When I open it in Codespaces, it doesn't work but locally it does. It seems that some other mac shortcuts take priority over the browser

Reply:
OK, I've added and icon here as the primary thing